league.js

```diff
@@ -0,0 +1,323 @@
+    // Refreshes the latest roster information
+    //--------------------------------------------------------------------------
+    'refreshRosters': function(callback) {
+    },
+
+    //--------------------------------------------------------------------------
+    // Refreshes the latest mods information
+    //--------------------------------------------------------------------------
+    'refreshMods': function(callback) {
+    },
+
+    //--------------------------------------------------------------------------
+    // Figures out the current scheduling information for the round.
+    //--------------------------------------------------------------------------
+    'refreshCurrentRoundSchedules': function(callback) {
+        var query_options = {
+            'min-row': 1,
+            'max-row': 100,
+            'min-col': 1,
+            'max-col': 8,
+            'return-empty': true
+        }
+        var self = this;
+        spreadsheets.get_rows(
+            self.options.spreadsheet.service_account_auth,
+            self.options.spreadsheet.key,
+            query_options,
+            function(err, rows) {
+                if (err) { return callback(err, rows); }
+                var new_pairings = [];
+                rows.forEach(function(row) {
+                    if (!row['white'].value || !row['black'].value) { return; }
+                    if (row['result'].formula) {
+                        var link = spreadsheets.parse_hyperlink(row['result'].formula || "");
+                    } else {
+                        var link = {'text': row['result'].value};
+                    }
+                    var date_string = row[self.options.spreadsheet.schedule_colname].value || '';
+                    date_string = date_string.trim()
+                    var date = moment.utc(
+                        moment.utc().year() + "/" + date_string,
+                        "YYYY/MM/DD @ HH:mm",
+                        true
+                    );
+                    if (!date.isValid()) {
+                        date = undefined;
+                    }
+                    new_pairings.push({
+                        white: self.canonical_username(row['white'].value),
+                        black: self.canonical_username(row['black'].value),
+                        result: link['text'],
+                        url: link['href'],
+                        scheduled_date: date
+                    });
+                });
+                self._pairings = new_pairings;
+                callback(undefined, self._pairings);
+            }
+        );
+    },
+
+    //--------------------------------------------------------------------------
+    // Finds the pairing for this current round given either a black or a white
+    // username.
+    //--------------------------------------------------------------------------
+    'findPairing': function(white, black) {
+        white = white || undefined;
```

Comment:
is this an idiom I am not familiar with?

Essentially, if white has a value that will evaluate to true, use it, otherwise assign undefined.

It will replace 0, false, null, etc with undefined.

Anything I missed? Why make this assignment?